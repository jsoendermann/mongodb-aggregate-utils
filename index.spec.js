const { lookupOne } = require('.')

describe('lookupOne', () => {
  it('should work with default params', () => {
    const stages = lookupOne({
      from: 'there',
      localField: 'ourKey',
      foreignField: 'theirKey',
      as: 'newField',
      arrayName: 'arrayName',
    })

    expect(stages).toMatchSnapshot()
  })

  it('should respect the keepArray flag', () => {
    const stages = lookupOne({
      from: 'there',
      localField: 'ourKey',
      foreignField: 'theirKey',
      as: 'newField',
      arrayName: 'arrayName',
      keepArray: true,
    })

    expect(stages).toHaveLength(2)
  })
})
