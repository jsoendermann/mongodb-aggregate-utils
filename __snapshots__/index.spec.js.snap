// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lookupOne should work with default params 1`] = `
Array [
  Object {
    "$lookup": Object {
      "as": "arrayName",
      "foreignField": "theirKey",
      "from": "there",
      "localField": "ourKey",
    },
  },
  Object {
    "$addFields": Object {
      "newField": Object {
        "$arrayElemAt": Array [
          "$arrayName",
          0,
        ],
      },
    },
  },
  Object {
    "$project": Object {
      "arrayName": 0,
    },
  },
]
`;
